src/tests/core/engine/processors/abstract_processor_test.py

```diff
@@ -0,0 +1,125 @@
+        test_class = ChildProcessor(app_engine_config)
+        self.assertEqual(test_class.app_processor_config.id, app_engine_config.id)
+
+    def test_app_engine_processor_context(self):
+        app_engine_processor_context = build_app_engine_processor_context([])
+        test_class = ChildProcessor(None, app_engine_processor_context)
+        self.assertEqual(
+            test_class.app_context.processors, app_engine_processor_context.processors
+        )
+
+    def test_self_pause_is_false(self):
+        test_class = ChildProcessor()
+        self.assertEqual(test_class.pause, False)
+
+    def test_self_destroy_is_false(self):
+        test_class = ChildProcessor()
+        self.assertEqual(test_class.destroy, False)
+
+    def test_last_execution_time(self):
+        app_engine_config = build_app_engine_config(id="test config", interval=10)
+        test_class = ChildProcessor(app_engine_config)
+        test_class.start()
+        self.assertAlmostEqual(test_class.last_execution_time, datetime.now())
+
+    def test_process_method_was_call_with(self):
+        app_engine_config = build_app_engine_config(id="test config", interval=10)
+        test_class = ChildProcessor(app_engine_config)
+
+        with patch.object(ChildProcessor, "process") as mock:
+            test_class.start()
+        mock.assert_called_with(test_class.delta_time)
+
+    def test_on_start_set_once(self):
+        mock_function = Mock(wraps=lambda: print("on started"))
+        app_engine_config = build_app_engine_config(
+            id="test config", interval=10, on_start=mock_function
+        )
+        test_class = ChildProcessor(app_engine_config)
+        test_class.start()
+        mock_function.assert_called_once()
+
+    def test_on_execute_set_everytime_process_run(self):
+        mock_function = Mock(wraps=lambda: print("on execute"))
+        app_engine_config = build_app_engine_config(
+            id="test config", interval=10, on_execute=mock_function
+        )
+        test_class = ChildProcessor(app_engine_config)
+        test_class.start()
+        mock_function.assert_called()
+
+    def test_thread_destroy(self):
+        app_engine_config = build_app_engine_config(id="test config", interval=10)
+        test_class = ChildProcessor(app_engine_config)
+        test_class.start()
+        self.assertTrue(test_class.is_alive())
+        test_class.destroy = True
+        time.sleep(15)
+        self.assertFalse(test_class.is_alive())
+
+    def test_on_destroy_when_destroy_set_true(self):
+        mock_function = Mock(wraps=lambda: print("on destroy"))
+        app_engine_config = build_app_engine_config(
+            id="test config", interval=10, on_destroy=mock_function
+        )
+        test_class = ChildProcessor(app_engine_config)
+        test_class.start()
+        self.assertTrue(test_class.is_alive())
+        test_class.destroy = True
+        time.sleep(15)
```

Comment:
hmm this is tricky, could you mock `time.sleep` implementation and control how much time sleep function will last? It is better for scalability and flexibility